sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient_private.cs

```diff
@@ -87,12 +85,56 @@ private Request CreateRequest(HttpPipelineMethod method, params string[] path)
             return request;
         }
 
+        private Request CreateRequest(HttpPipelineMethod method, Uri uri)
+        {
+            Request request = _pipeline.CreateRequest();
+
+            request.Headers.Add(HttpHeader.Common.JsonContentType);
+            request.Headers.Add(HttpHeader.Common.JsonAccept);
+            request.Method = method;
+            request.UriBuilder.Uri = uri;
+
+            return request;
+        }
+
+        private Uri CreateFirstPageUri(string path)
```

Comment:
This doesn't really help, to be efficient `AppendQuery` needs to be called on `request.UriBuilder` but it might require way too much refactoring for this PR.